scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GoalBuilder.java

```diff
@@ -48,11 +48,29 @@ public static Goal goalOfGoalSpecifier(
               g.forEach(),
               lookupActivityType))
           .thereExistsOne(makeActivityTemplate(g.activityTemplate(), lookupActivityType));
-      // TODO: This if statement should be removed when the DSL supports time expressions.
-      if (g.forEach() instanceof SchedulingDSL.ConstraintExpression.ActivityExpression) {
-        builder = builder.startsAt(TimeAnchor.END);
-      } else {
-        builder = builder.startsAt(TimeAnchor.START);
+
+      if (g.startConstraint().isPresent()) {
+        final var startConstraint = g.startConstraint().get();
+        final var timeExpression = new TimeExpressionRelativeFixed(
+            startConstraint.windowProperty(),
+            startConstraint.singleton(),
+            "George");
+        timeExpression.addOperation(startConstraint.operator(), startConstraint.operand());
+        builder.startsAt(timeExpression);
+      }
+
+      if (g.endConstraint().isPresent()) {
+        final var startConstraint = g.endConstraint().get();
+        final var timeExpression = new TimeExpressionRelativeFixed(
+            startConstraint.windowProperty(),
+            startConstraint.singleton(),
+            "George");
```

Comment:
Is `"George"` supposed to be here? At first I thought this was some test file – maybe there's context I'm missing?

Reply:
I'd like to lobby for this to be `"Paul"`

Reply:
😆 Really good point, this was me being amused that there's a useless field. A lot of nodes in the scheduler have name fields, perhaps for debugging purposes, but we should probably consider eliminating them.

I've been operating off of the Weasleys, but I'd be happy to pivot to the Beatles 😝

Reply:
Awesome, thanks Matt :)